appscale/tools/agents/azure_agent.py

```diff
@@ -446,13 +453,34 @@ def run_instances(self, count, parameters, security_configured, public_ip_needed
     credentials = self.open_connection(parameters)
     subscription_id = str(parameters[self.PARAM_SUBSCRIBER_ID])
     virtual_network = parameters[self.PARAM_GROUP]
+    resource_group = parameters[self.PARAM_RESOURCE_GROUP]
 
     network_client = NetworkManagementClient(credentials, subscription_id)
+    compute_client = ComputeManagementClient(credentials, subscription_id)
+
     subnet = self.create_virtual_network(network_client, parameters,
                                          virtual_network, virtual_network)
 
     active_public_ips, active_private_ips, active_instances = \
       self.describe_instances(parameters)
+
+    # Create an Availability Set for Load Balancer VMs
+    lb_avail_set_name = "lb-availability-set"
```

Comment:
Wouldn't we want want different deployments to use different availability sets?

Reply:
So when we create an availability set, it is basically similar to having a folder for all the load balancer VMs to be placed within and then those VMs are removed from it when the deployment is terminated. I was of the opinion that the same availability set can just be reused between different deployments, because their only use right now is to have a set of load balancer VMs which then can be used by the Azure load balancer to distribute load to.

I can use a lib to automatically generate a new name instead of hard coding it here but I thought this name would be more relevant because we would only have one per deployment.

Reply:
So if you have two deployments running at the same time and they are both using the same availability set, how does the load balancer know which VMs should get which traffic?

Reply:
Oh so this is per resource group in Azure. And we can currently only do one deployment per resource group in Azure. I meant the same availability set name to be used between deployments started and terminated within the same resource group. Outside of the resource group, even if there exists another same named availability set within another resource group it wouldn't matter, because all the deployment resources are bound with the resource group under within they are created. So there would not be any clashes.

Reply:
Thanks, that makes sense.